### PR TITLE
Added hints to each quiz question

### PR DESCRIPTION
**Pull request by rw251**
Also:
- my prettifier has tidied up the setting.json file, including removing a trailing comma
- debug renamed to show
- updated the wording in q8 to make something i found confusing a bit more explicit

### CODE REVIEW

File: quiz_answers.py [hunk 1]
```diff
@@ -219,6 +265,12 @@
     earliest_diagnosis_date.is_not_null()
     & earliest_diagnosis_date.is_on_or_between("2023-04-01", "2024-03-31")
 )
+questions[3]._hint = (
+    "This builds on question 1. Assign your answer to question 1 to a variable like this:\n\n"
+    "earliest_diabetes_diagnosis = ...your answer to q1...\n\n"
+    "You can then use this new variavle in conjunction with one of the date functions e.g. "
```
Comment (madwort) [inline quiz_answers.py:271]:
```suggestion
    "You can then use this new variable in conjunction with one of the date functions e.g. "
```

File: quiz_answers.py [hunk 2]
```diff
@@ -249,6 +301,12 @@
     """
 )
 questions[4].expected = (earliest_referral_date - earliest_diagnosis_date).months
+questions[4]._hint = (
+    "It might be helpful to assign the answers to q1 and q2 into two variables called "
+    "earliest_diabetes_diagnosis and earliest_referral. You can then find the difference "
+    "between these two dates. Subtracting dates leads to a 'DateDifference', so you will also "
```
Comment (madwort) [inline quiz_answers.py:307]:
```suggestion
    "between these two dates. Subtracting dates returns a 'DateDifference', so you will also "
```

File: quiz_answers.py [hunk 3]
```diff
@@ -365,6 +432,17 @@
     .date
 )
 questions[7].expected = latest_moderate_or_severe_frailty_date
+questions[7]._hint = (
+    (
+        "You'll need to either:\n\n"
+        " - combine the moderate and severe frailty codelists, which can be done like this: "
+        "moderate_frailty_codes + severe_frailty_codes\n"
+        " - or make use of the 'or' operator '|' to find snomedct_codes in "
+        "moderate_frailty_codes OR in severe_frailty_codes."
```
Comment (madwort) [inline quiz_answers.py:441]: hmmm, so both `....is_in(moderate + severe)` and `....is_in(moderate) | ....is_in(severe)` are valid? what about `....is_in(moderate | severe)`?

Comment (madwort) [reply quiz_answers.py:441]: anyway, I came here to say I'd like to see the or operator used in the example, but then when I try it's basically not possible to do that in a good way, so... 🤷 

```suggestion
        " - combine the moderate and severe frailty codelists, which can be done like this: "
        "moderate_frailty_codes + severe_frailty_codes\n"
        " - or make use of the 'or' operator '|' to find snomedct_codes in "
        "is_in(moderate_frailty_codes) | is_in(severe_frailty_codes)"
```

File: quiz_answers.py [hunk 4]
```diff
@@ -404,6 +482,17 @@
     )
 )
 questions[8].expected = has_moderate_or_severe_frailty
+questions[8]._hint = (
+    (
+        "This is a bit complicated, but essentially we're trying to match one of the following:\n"
+        "1. A patient with a moderate or severe frailty code AND no mild frailty code, or\n"
+        "2. A patient with a moderate or severe frailty code, AND a mild frailty code, but where "
+        "the most recent frailty code is the moderate/severe one\n"
+        "You'll need the is_null() and is_not_null() methods."
```
Comment (madwort) [inline quiz_answers.py:491]:
```suggestion
        "the most recent frailty code is the moderate/severe one.\n"
        "You'll also need the is_null() and is_not_null() methods."
```

File: quiz_answers.py (hunk 3 above)
Comment (rw251) [reply quiz_answers.py:441]: I guess I was trying to not spoon feed the answer, so tell people about the operators, then tell them what to do in pseudo code and let them figure out the rest.

Comment (madwort) [reply quiz_answers.py:441]: yep fine

**Pull request merged**